functions/Export-DbaAvailabiltyGroup.ps1

```diff
@@ -0,0 +1,173 @@
+        }
+
+        if ($AllAGs.count -eq 0) {
+            Write-Output "No Availability Groups detected on '$SqlServer'"
+        }
+
+        # Set and create the OutputLocation if it doesn't exist
+        $SQLINST = $SQLServer.Replace('\','$')
+        $OutputLocation = "${FilePath}\${SQLINST}"
+
+        if (!(Test-Path $OutputLocation -PathType Container)) {
+            New-Item -Path $OutputLocation -ItemType Directory -Force | Out-Null
+        } 
+
+        # Script each Availability Group
+        foreach ($ag in $AllAGs) {
+            $AGName = $ag.Name
+
+            # Set the outfile name
+            if ($AppendDateToOutputFilename.IsPresent) {
+                $Dttm = (Get-Date -Format 'yyyyMMdd_hhmm')
+                $OutFile = "${OutputLocation}\${AGname}_${Dttm}.sql"
+            } else {
+                $OutFile = "${OutputLocation}\${AGname}.sql"
+            }
+
+            # Check NoClobber and script out the AG
+            if ($NoClobber.IsPresent -and (Test-Path -Path $OutFile -PathType Leaf)) {
+                Write-Warning "OutputFile '$OutFile' already exists. Skipping due to -NoClobber parameter"
+            } else {
+                Write-output "Scripting Availability Group [$AGName] on [$SQLServer] to '$OutFile'"
+
+                # Create comment block header for AG script
+                "/*" | Out-File -FilePath $OutFile -Encoding ASCII -Force
+                " * Created by dbatools 'Export-DbaAvailabilityGroup' cmdlet on '$(Get-Date)'" | Out-File -FilePath $OutFile -Encoding ASCII -Append
+                " * See https://dbatools.io/Export-DbaAvailabilityGroup for more help" | Out-File -FilePath $OutFile -Encoding ASCII -Append
+
+                # Output AG and listener names
+                " *" | Out-File -FilePath $OutFile -Encoding ASCII -Append
+                " * Availability Group Name: $($ag.name)" | Out-File -FilePath $OutFile -Encoding ASCII -Append
+                $ag.AvailabilityGroupListeners | % {" * Listener Name: $($_.name)"} | Out-File -FilePath $OutFile -Encoding ASCII -Append
+
+                # Output all replicas
+                " *" | Out-File -FilePath $OutFile -Encoding ASCII -Append
+                $ag.AvailabilityReplicas | % {" * Replica: $($_.name)"} | Out-File -FilePath $OutFile -Encoding ASCII -Append
+
+                # Output all databases
+                " *" | Out-File -FilePath $OutFile -Encoding ASCII -Append
+                $ag.AvailabilityDatabases | % {" * Database: $($_.name)"} | Out-File -FilePath $OutFile -Encoding ASCII -Append
+
+                # $ag | Select-Object -Property * | Out-File -FilePath $OutFile -Encoding ASCII -Append
+                
+                "*/" | Out-File -FilePath $OutFile -Encoding ASCII -Append
+
+                # Script the AG
+                $ag.Script() | Out-File -FilePath $OutFile -Encoding ASCII -Append
+            }
+        }
+    }
+
+    END
+    {
+		$SQLObj.ConnectionContext.Disconnect()
```

Comment:
If you decide to support multiple servers (not so sure that's a good idea for this particular command), please move the disconnect up to within the loop